test/models/auth_session_test.rb

```diff
@@ -112,6 +112,18 @@ def setup
 
       @connector_mock.verify
     end
+
+    test "raises an error due to invalid creds" do
+      assert_raises(Exception) do
```

Comment:
API throws the `Exception` class error on failure

Reply:
Should we assert a more specific error? It could break the test suite in the future if Rails maintainers decide to change the error format but at least we are asserting correctly that the raised exception is the one from the on creation hook.

Reply:
I had the same thought, that's why I tried to be as specific as possible here:
https://github.com/EduardoSimon/budget.io/blob/b098589f7938774b3d8671e6f2245b77c5502c95/test/models/auth_session_test.rb#L116-L136

However, I saw in the test suite we don't have any env set for the credentials:
![image](https://github.com/EduardoSimon/budget.io/assets/59338032/83c06c39-2670-4c99-9412-3d2cb4bfcade)

That's why I came up with this test. We could assert with the more specific error but API only raises an error of Exception class.
What do you suggest?

Reply:
Maybe mocking the API call and manually raise a `StandardError` in the mocked call with the message which the API throws for the invalid creds?

Reply:
Agreed, we should mock that API call, could you please do that?

Reply:
Sure, let me do that and get back to you👍🏻